app/classifier/tasks/mobile-section.spec.js

```diff
@@ -0,0 +1,412 @@
+    let wrapper;
+
+    const mobileEnabledWorkflow = {
+      tasks: {
+        T0: {
+          answers: [
+            { label: 'Yes' },
+            { label: 'No' }
+          ]
+        }
+      },
+      configuration: {
+        swipe_enabled: true
+      }
+    };
+
+    before(function () {
+      wrapper = mount(<MobileSection
+        task={defaultTask}
+        workflow={mobileEnabledWorkflow}
+        project={defaultProject}
+      />);
+    });
+
+    it('checkbox is checked if mobile is enabled', function () {
+      assert.equal(wrapper.find('input').props().disabled, false);
+    });
+  });
+
+  describe('executes method when checked', function () {
+    let wrapper;
+
+    beforeEach(function () {
+      wrapper = mount(<MobileSection
+        task={defaultTask}
+        workflow={defaultWorkflow}
+        project={defaultProject}
+      />);
+    });
+
+    it('should call toggleShortcut with an input toggle', function () {
+      const toggleStub = sinon.stub(wrapper.instance(), 'toggleSwipeEnabled');
+      wrapper.update();
+      wrapper.find('input').simulate('change');
+      sinon.assert.called(toggleStub);
+    });
+  });
+
+  describe('only renders for questions (single task type)', function () {
+    let wrapper;
+
+    const singleTypeTask = {
+      question: 'What is it',
+      type: 'single',
+      answers: [
+        { label: 'Yes' },
+        { label: 'No' }
+      ]
+    };
+
+    before(function () {
+      wrapper = shallow(<MobileSection
+        task={singleTypeTask}
+        workflow={defaultWorkflow}
+        project={defaultProject}
+      />);
+    });
+
+    it('does not disable checkbox', function () {
+      assert.equal(wrapper.find('input').props().disabled, false);
+    });
+  });
+
+  describe('only renders for questions (multi task type)', function () {
+    let wrapper;
+
+    const multipleTypeTask = {
+      question: 'What is it',
+      type: 'multiple',
+      answers: [
+        { label: 'Yes' },
+        { label: 'No' }
+      ]
+    };
+
+    before(function () {
+      wrapper = shallow(<MobileSection
+        task={multipleTypeTask}
+        workflow={defaultWorkflow}
+        project={defaultProject}
+      />);
+    });
+
+    it('does not disable checkbox', function () {
+      assert.equal(wrapper.find('input').props().disabled, false);
+    });
+  });
+
+  describe('does not render for other task types', function () {
+    let wrapper;
+
+    const otherTypeTask = {
+      question: 'What is it',
+      type: 'blurgh',
+      answers: [
+        { label: 'Yes' },
+        { label: 'No' }
+      ]
+    };
+
+    before(function () {
+      wrapper = shallow(<MobileSection
+        task={otherTypeTask}
+        workflow={defaultWorkflow}
+        project={defaultProject}
+      />);
+    });
+
+    it('should not render for other task type', function () {
+      assert.equal(wrapper.find('input').props().disabled, false);
```

Comment:
This is a bit odd looking because it's the same check for the above 'does not disable checkbox'. Checking for a particular prop on an item that shouldn't exist seems counter intuitive.